### ISSUE TITLE
V8.0.1 - recycle bin / restoration problem

### ISSUE THREAD

**Issue by filefontaine** [umbraco/Umbraco-CMS#5328]
When you delete a node ( for me the root node) and you restore it, the children cannot be published.

Reproduction
------------

### Bug summary

publish node error (see screen below)

### Specifics

It's like a node is between the node and the child

### Steps to reproduce

* delete a node with multiple children, and this children have children
* restore the root node
* publish only the root node and try to publish a child or a child of a child

### Expected result

the child node be published

### Actual result

![Screenshot_2019-04-24 Content - localhost](https://user-images.githubusercontent.com/22192319/56640137-867aa400-6672-11e9-88df-8c6bd52e0f08.png)

![Screenshot_2019-04-24 Content - localhost(1)](https://user-images.githubusercontent.com/22192319/56640182-a4e09f80-6672-11e9-8778-527185038159.png)

(Accueil and Etudes is published but i cant publish Master)

**Comment by nul800sebastiaan** [comment]
I can reproduce this one, it works on the second-level child (for me) but not on the 3 level.

So "About this starter kit" publishes just fine, but "Test" under that doesn't (neither does "Test2").

![image](https://user-images.githubusercontent.com/304656/56722022-759b6280-6746-11e9-9e58-e07be9189386.png)

Even after using "Publish with descendants" on "About Us" I can't publish "Test" nor "Test2", even though I can unpublish them and publish them using "Publish with descendants".

**Comment by zpqrtbnk** [comment]
Reproduced.

Details: 
* deleted the root document, then restored it. The restored root document is unpublished. All documents underneath are "published but not visible" because a document up in their tree branch is not published - so far, it's all normal
* "saved and published" the root document. Now it is published. And all documents underneath become visible again, one can navigate to them, they have a url - again, so far, it's all normal
* "saved and published" a 2nd level child - works as expected. "saved and published" a 3rd level child - and get the error message "Publish: ... can not be published, because a parent page is not published."

Investigating

**Comment by zpqrtbnk** [comment]
That is the database after the restore:

![image](https://user-images.githubusercontent.com/2846429/56736357-3d0c8080-6768-11e9-9d82-8717f4036012.png)

The `level` there are wrong, so we have an issue with levels when moving things around / restoring things.

**Comment by zpqrtbnk** [comment]
One can fix the levels with:
~~~~
update umbracoNode
set level = len(path) - len(replace(path, ',', ''))
where nodeObjectType = 'C66BA18E-EAF3-4CFF-8A22-41B16D66A972'
~~~~
And then everything works again.

Now on to figuring out what is corrupting levels.

**Comment by zpqrtbnk** [comment]
Oh but... *this* is the database after moving to the Recycle Bin...

![image](https://user-images.githubusercontent.com/2846429/56737210-33841800-676a-11e9-9023-47ced94b816b.png)

So it's not so much moving *back* (restoring) that does weird things: moving *to* the recycle bin breaks levels!

**Comment by zpqrtbnk** [comment]
But, if I fix the levels in the bin and restore, it breaks again, so it's moving "in general" that is broken.

**Comment by zpqrtbnk** [comment]
PR #5347 

Review = now, should be possible to move to and from recycle bin without problems.

**_Note to Reviewer: this is an 8.0.2 fix. Merge to v8/dev and then cherry pick to v8/8.0._**

**Comment by nul800sebastiaan** [comment]
Merged to 8.0.2 in 1b5d614f5ae794f5c27af2222146315e6286d973
